### PR TITLE
Fixing post param with no value

### PR DESCRIPTION
**Pull request by djrenfro**
Fixing scenario where there is no value after an '=' in a post param

### CODE REVIEW

File: proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java [hunk 1]
```diff
@@ -184,6 +184,9 @@ public static String removePortFromHostHeaderString(String host) {
                     }
 
                     values.add(value.toString());
+                    if (equalsPos != -1 && value.size() == 0) {
```
Comment (dvanderb) [inline proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java]: add comments

File: proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java [hunk 2]
```diff
@@ -409,10 +412,13 @@ public static void handleStandardPost(EntityEnclosingMethod methodProxyRequest,
                     }
                 }
             }
-            String postData = new Proxy().processPostDataString(requestBody.toString());
-            requestBody = new StringBuilder(postData);
-            requestByteArray = postData.getBytes();
-            requestEntity = new ByteArrayRequestEntity(requestByteArray);
+            Proxy.QueryInformation queryInformation = Proxy.processPostDataString(requestBody.toString());
```
Comment (dvanderb) [inline proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java]: add comments

File: proxyserver/src/main/java/com/groupon/odo/Proxy.java [hunk 3]
```diff
@@ -1296,4 +1300,9 @@ private String stripJSONPToOutstr(HttpServletRequest httpServletRequest, PluginR
         public HttpRequestInfo originalRequestInfo = null;
         public String jsonpCallback = null;
     }
+
```
Comment (dvanderb) [inline proxyserver/src/main/java/com/groupon/odo/Proxy.java]: comment class to say what it is used for

Comment (djrenfro) [inline proxyserver/src/main/java/com/groupon/odo/Proxy.java]: Comment added

File: proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java (hunk 2 above)
Comment (djrenfro) [inline proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java]: added

File: proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java (hunk 1 above)
Comment (djrenfro) [inline proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java]: added

File: proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java [hunk 4]
```diff
@@ -409,10 +417,20 @@ public static void handleStandardPost(EntityEnclosingMethod methodProxyRequest,
...
-            requestBody = new StringBuilder(postData);
-            requestByteArray = postData.getBytes();
-            requestEntity = new ByteArrayRequestEntity(requestByteArray);
+            /**
+             * Process the post data string so it can be added to history
+             * Separates individual params out and applies post data override as applicable
+             */
+            Proxy.QueryInformation queryInformation = Proxy.processPostDataString(requestBody.toString());
+            // Set request body which is added to history
+            requestBody = new StringBuilder(Proxy.processPostDataString(requestBody.toString()).queryString);
```
Comment (dwillson) [inline proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java]: Need to reprocess here, or can you use queryInformation.queryString?

Comment (djrenfro) [inline proxyserver/src/main/java/com/groupon/odo/HttpUtilities.java]: Fixed

**Comment by dwillson** [comment]
:+1:

**Pull request merged**